pyomo/contrib/viewer/tests/test_data_model_item.py

```diff
@@ -38,18 +44,62 @@ def blackbox(a,b):
         m.c1 = Constraint(expr=m.x[0] * m.z[0]**2 + self.bb(m.x[0],m.x[1]) == 2*sqrt(2.0))
         m.c2 = Constraint(expr=m.z[2]**4 * m.z[1]**2 + m.z[1] == 8+sqrt(2.0))
         m.c3 = Constraint(expr=m.x[1] == 3)
+        m.c4 = Constraint(expr=0 == 3/m.x[2])
+        m.c5 = Constraint(expr=0 == log(m.x[2]))
+        m.c6 = Constraint(expr=0 == log(m.x[2]-4))
+        m.c7 = Constraint(expr=0 == log(m.x[3]))
         self.m = m.clone()
 
     def test_expr_calc(self):
         cdi = ComponentDataItem(parent=None, ui_setup=None, o=self.m.b1.e1)
         cdi.calculate()
         assert(abs(cdi.get("value")-3) < 0.0001)
 
+    def test_expr_calc_div0(self):
+        cdi = ComponentDataItem(parent=None, ui_setup=None, o=self.m.b1.e2)
+        cdi.calculate()
+        assert(cdi.get("value") == "Divide_by_0")
```

Comment:
I am not going to block this PR (as the tests are still valid), but in the future, the "unittest way" to assert equality is:
```
self.assertEqual(cdi.get("value"), "Divide_by_0")
```

Reply:
I may as well update these, while I'm fixing things related to the other comment.